.github/workflows/container-image.yml

```
@@ -1,262 +1,117 @@
+      -
+        name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
-      - name: Login to GitHub Container Registry
+      -
+        name: Login to GitHub Container Registry
         uses: docker/login-action@v2
         with:
           registry: ghcr.io
           username: ${{ github.repository_owner }}
           password: ${{ secrets.GITHUB_TOKEN }}
-
-      # Build and push `latest` image
-      - name: Build and export to docker
-        if: ${{ startsWith(github.ref, 'refs/heads/') }}
-        uses: docker/build-push-action@v4
-        with:
-          context: .
-          load: true
-          tags: |
-            ghcr.io/${{ github.repository_owner }}/policy-server:latest
-      - name: Test container image
-        if: ${{ startsWith(github.ref, 'refs/heads/') }}
-        run: |
-          docker run --rm ghcr.io/${{ github.repository_owner }}/policy-server:latest --help
-      - name: Build and push
-        if: ${{ startsWith(github.ref, 'refs/heads/') }}
-        id: build-latest
-        uses: docker/build-push-action@v4
-        with:
-          context: .
-          platforms: linux/amd64, linux/arm64
-          push: true
-          tags: |
-            ghcr.io/${{ github.repository_owner }}/policy-server:latest
-
-      # Build and push the `:<version>` image
-      - name: Retrieve tag name (main branch)
+      -
+        name: Install cosign
+        if: ${{ inputs.push-image }}
+        uses: sigstore/cosign-installer@v3
+      -
+        name: Retrieve tag name (main branch)
         if: ${{ startsWith(github.ref, 'refs/heads/main') }}
         run: |
           echo TAG_NAME=latest >> $GITHUB_ENV
-      - name: Retrieve tag name (feat branch)
+      -
+        name: Retrieve tag name (feat branch)
         if: ${{ startsWith(github.ref, 'refs/heads/feat') }}
         run: |
           echo "TAG_NAME=latest-$(echo ${GITHUB_REF#refs/heads/})" >> $GITHUB_ENV
-      - name: Retrieve tag name (tag)
+      -
+        name: Retrieve tag name (tag)
         if: ${{ startsWith(github.ref, 'refs/tags/') }}
         run: |
-          echo TAG_NAME=$(echo $GITHUB_REF | sed -e "s|refs/tags/||") >> $GITHUB_ENV
-      - name: Build and export to docker
-        if: ${{ startsWith(github.ref, 'refs/tags/') }}
+            echo TAG_NAME=$(echo $GITHUB_REF | sed -e "s|refs/tags/||") >> $GITHUB_ENV
+      -
+        name: Build and push container image
+        if: ${{ inputs.push-image }}
+        id: build-image
         uses: docker/build-push-action@v4
         with:
           context: .
-          load: true
+          file: ./Dockerfile
+          platforms: linux/amd64, linux/arm64
+          push: true
           tags: |
-            ghcr.io/${{ github.repository_owner }}/policy-server:${{ env.TAG_NAME }}
-      - name: Test container image
-        if: ${{ startsWith(github.ref, 'refs/tags/') }}
+            ghcr.io/${{github.repository_owner}}/policy-server:${{ env.TAG_NAME }}
```

Just a realization:

```suggestion
            ghcr.io/${{github.repository_owner}}/{{ github.event.repository.name }}:${{ env.TAG_NAME }}
```

(and elsewhere)
And at this point, the workflows are the same between repos. I don't want to do that right now though.

---

Unfortunately they are no longer the same, since now we are back to cross compilation :disappointed:

---

Marking as solved, unfortunately there's no solution right now. We have to wait for GitHub to offer native arm64 runners